HIPs/firefly.md

```diff
@@ -0,0 +1,377 @@
+
+## Context
+
+Enterprise blockchain emerged in 2015 as a new type of B2B collaboration software, helping companies to exchange data with common business logic in _smart contracts_ while having a common record of transactions with a* shared ledger.* Enterprise blockchain use cases see companies organizing into business networks with each company deploying a private copy of the software stack including a blockchain node. The nature of a decentralized application with many independent parties means that sophisticated coordination across the parties is critical for end to end flows to execute smoothly. Multi-party flows necessarily achieve that coordination via event-driven designs including application level messaging and the triggering of actions from blockchain level event streams.
+
+Further, most enterprise use cases require privacy within a given transaction - the exposure of data to a subset of members in the network on a “need to know” basis. This combined with the immutability of the shared ledger means that many use cases center their data flows in off chain interactions.
+
+For these reasons, we have seen that enterprise blockchain use cases require a set of off chain technologies that serve as “plumbing” layers to reliably coordinate and facilitate the multi-party data flows. Rationalizing identity, connectivity and synchronization across these layers becomes a core challenge to building a blockchain project.
+
+We’ve observed that the specific application and middleware components and how they are used have a remarkable number of similarities regardless of the industry, use case, or even blockchain protocol. Kaleido has six years of learnings on these use cases across Fabric, Ethereum, and Corda and has built these components out over a period of several years of active engineering. Hyperledger FireFly is born out of those learnings and engineering work. The code contribution is a direct refactoring of core Kaleido components which are run in production by large enterprises today.
+
+## Dependent Projects
+
+Hyperledger FireFly requires a blockchain protocol to run. FireFly leverages two fundamental aspects of the blockchain protocol for its integrity - data immutability and global ordering. At time of proposal, full support is included for is included for Enterprise Ethereum variants including Quorum and **Hyperledger Besu**.  Support for **Hyperledger Fabric** is in active design and planned to be implemented.  Support for Corda was available in the prior version of code (called KAT), but requires additional design and implementation to port to FireFly.  The priority for Corda support will be set by the FireFly community.  
```

Comment:
`full support is included for is included` --> `full support is included`

Reply:
Thanks Tracy, updated